ledger/src/shred.rs

```diff
@@ -247,6 +247,13 @@ impl Shred {
         packet.meta.size = len;
     }
 
+    pub fn copy_from_packet(packet: &Packet) -> Result<Self> {
+        let mut serialized_shred = vec![0; SHRED_PAYLOAD_SIZE];
+        // TODO: assert packet.data.len() >= SHRED_PAYLOAD_SIZE / == PACKET_DATA_SIZE ?
```

Comment:
Thoughts on this assertion? We had something like this in window service; we could move it here to avoid repeated lines for all the callers of this.